gcs/src/main/java/com/google/cloud/hadoop/fs/gcs/GhfsThreadLocalStatistics.java

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2025 Google Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.hadoop.fs.gcs;
+
+import static com.google.cloud.hadoop.fs.gcs.GhfsStatistic.GCS_CONNECTOR_TIME;
+
+import com.google.cloud.hadoop.gcsio.GoogleCloudStorageStatistics;
+import java.util.*;
+import org.apache.hadoop.fs.StorageStatistics;
+
+class GhfsThreadLocalStatistics extends StorageStatistics {
+  static final String NAME = "GhfsThreadLocalStatistics";
+  static final String HADOOP_API_COUNT = "hadoopApiCount";
+  static final String HADOOP_API_TIME = "hadoopApiTime";
+  static final String GCS_API_COUNT = "gcsApiCount";
+  static final String GCS_API_TIME = "gcsApiTime";
+  static final String BACKOFF_COUNT = "backoffCount";
+  static final String BACKOFF_TIME = "backoffTime";
+
+  private final ThreadLocalValue hadoopApiCount;
+  private final ThreadLocalValue hadoopApiTime;
+  private final ThreadLocalValue gcsApiCount;
+  private final ThreadLocalValue gcsApiTime;
+  private final ThreadLocalValue backoffCount;
+  private final ThreadLocalValue backoffTime;
+
+  private Map<String, ThreadLocalValue> metrics = new HashMap<>();
+
+  GhfsThreadLocalStatistics() {
+    super(NAME);
+    this.hadoopApiCount = createMetric(HADOOP_API_COUNT);
+    this.hadoopApiTime = createMetric(HADOOP_API_TIME);
+    this.gcsApiCount = createMetric(GCS_API_COUNT);
+    this.gcsApiTime = createMetric(GCS_API_TIME);
+    this.backoffCount = createMetric(BACKOFF_COUNT);
+    this.backoffTime = createMetric(BACKOFF_TIME);
+  }
+
+  private ThreadLocalValue createMetric(String name) {
+    ThreadLocalValue result = new ThreadLocalValue();
+    metrics.put(name, result);
+
+    return result;
+  }
+
+  @Override
+  public Long getLong(String s) {
+    if (!metrics.containsKey(s)) {
+      return 0L;
+    }
+
+    return metrics.get(s).getValue();
+  }
+
+  @Override
+  public boolean isTracked(String s) {
+    return metrics.containsKey(s);
+  }
+
+  @Override
+  public void reset() {
+    for (ThreadLocalValue s : metrics.values()) {
+      s.reset();
+    }
+  }
+
+  void increment(GhfsStatistic statistic, long count) {
+    if (statistic == GCS_CONNECTOR_TIME) {
+      this.hadoopApiTime.increment(count);
```

Comment:
Sorry, I am not able to understand this, why are we incrementing hadoopApiTime for GCS_CONNECTOR_TIME?

Also, the below metric (2 lines below) is Count while here it is Time.

Reply:
GCS connector implements Hadoop FS APIs. So we are measuring the wall clock time taken by GcsConnectorHadoopAPIs. 

Another way to look at this is: From the caller perspective this the "hadoopAPITime" and from the connector perspective it is the GCSConnectorTime. Also, from the caller perspective they need the metric to be generic since the connector could be for a different scheme.